src/test/java/seedu/planner/model/course/MajorDescriptionTest.java

```diff
@@ -0,0 +1,83 @@
+package seedu.planner.model.course;
+
+import static org.junit.jupiter.api.Assertions.assertEquals;
+import static org.junit.jupiter.api.Assertions.assertFalse;
+import static org.junit.jupiter.api.Assertions.assertTrue;
+
+import java.io.IOException;
+import java.util.ArrayList;
+import java.util.HashMap;
+import java.util.List;
+import java.util.Map;
+
+import org.junit.Rule;
+import org.junit.Test;
+import org.junit.rules.TemporaryFolder;
+
+import seedu.planner.commons.util.JsonUtil;
+
+public class MajorDescriptionTest {
+    @Rule
+    public TemporaryFolder testFolder = new TemporaryFolder();
+
+    /**
+     * Create dummy Major Description object to test serialization and deserialization.
+     */
+    private MajorDescription createDummyMajorDescription() {
+        List<ModuleDescription> modules = new ArrayList<>();
+        modules.add(new ModuleDescription("CS1010", ProgrammeRequirement.FOUNDATION));
+        modules.add(new ModuleDescription("CS1231", ProgrammeRequirement.FOUNDATION));
+        modules.add(new ModuleDescription("CS2030", ProgrammeRequirement.FOUNDATION));
+        modules.add(new ModuleDescription("CS2040", ProgrammeRequirement.FOUNDATION));
+        modules.add(new ModuleDescription("CS2100", ProgrammeRequirement.FOUNDATION));
+        modules.add(new ModuleDescription("CS2103T", ProgrammeRequirement.BREATH_AND_DEPTH,
+                List.of(FocusArea.SOFTWARE_ENGINEERING)));
+        modules.add(new ModuleDescription("CS2105", ProgrammeRequirement.FOUNDATION));
+        modules.add(new ModuleDescription("CS2106", ProgrammeRequirement.FOUNDATION));
+        modules.add(new ModuleDescription("CS3230", ProgrammeRequirement.INDUSTRIAL_EXPERIENCE_REQUIREMENT));
+        modules.add(new ModuleDescription("CP3880", ProgrammeRequirement.INDUSTRIAL_EXPERIENCE_REQUIREMENT));
+        modules.add(new ModuleDescription("CP3200", ProgrammeRequirement.INDUSTRIAL_EXPERIENCE_REQUIREMENT));
+        modules.add(new ModuleDescription("CP3202", ProgrammeRequirement.INDUSTRIAL_EXPERIENCE_REQUIREMENT));
+        modules.add(new ModuleDescription("IS4010", ProgrammeRequirement.INDUSTRIAL_EXPERIENCE_REQUIREMENT));
+        modules.add(new ModuleDescription("CP3200", ProgrammeRequirement.INDUSTRIAL_EXPERIENCE_REQUIREMENT));
+        modules.add(new ModuleDescription("CP3107", ProgrammeRequirement.INDUSTRIAL_EXPERIENCE_REQUIREMENT));
+
+        modules.add(new ModuleDescription("IS1103", ProgrammeRequirement.IT_PROFESSIONALISM));
```

Comment:
Perhaps can move this line up to be consistent with the rest. Or we can separate the lines based on their module type

Reply:
Done. Moved the line up.